01.md

```diff
@@ -1,177 +1,235 @@
-    ["a", "30023:f7234bd4c1394dda46d09f35bd384dd30cc552ad5541990f98844fb06676e9ca:abcd", "wss://nostr.example.com"],
-    ["alt", "reply"],
-    ...
-  ],
-  ...
-}
-```
+To verify the authenticity of an event: 
+1. Recalculate the `.id` and check against the incoming `.id`
+2. Verify if the signature is valid for the `.id` and `.pubkey`. 
 
-The first element of the tag array is referred to as the tag _name_ or _key_ and the second as the tag _value_. So we can safely say that the event above has an `e` tag set to `"5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36"`, an `alt` tag set to `"reply"` and so on. All elements after the second do not have a conventional name.
+## Kinds
 
-This NIP defines 3 standard tags that can be used across all event kinds with the same meaning. They are as follows:
+The `.kind` attribute specifies the meaning of an event, its tags and it is further defined in the various NIPs in this repository
 
-- The `e` tag, used to refer to an event: `["e", <32-bytes lowercase hex of the id of another event>, <recommended relay URL, optional>]`
-- The `p` tag, used to refer to another user: `["p", <32-bytes lowercase hex of a pubkey>, <recommended relay URL, optional>]`
-- The `a` tag, used to refer to a (maybe parameterized) replaceable event
-    - for a parameterized replaceable event: `["a", <kind integer>:<32-bytes lowercase hex of a pubkey>:<d tag value>, <recommended relay URL, optional>]`
-    - for a non-parameterized replaceable event: `["a", <kind integer>:<32-bytes lowercase hex of a pubkey>:, <recommended relay URL, optional>]`
+Specific ranges define storage behaviors: 
 
-As a convention, all single-letter (only english alphabet letters: a-z, A-Z) key tags are expected to be indexed by relays, such that it is possible, for example, to query or subscribe to events that reference the event `"5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36"` by using the `{"#e": "5c83da77af1dec6d7289834998ad7aafbd9e2191396d75ec3cc27f5a77226f36"}` filter.
+| Name          | Range                             | SHOULD retain                                          |
+| ------------- | --------------------------------- | ------------------------------------------------------ |
+| Regular       | ` 1000 <= kind < 10000`           | All events                                             |
+| Replaceable   | `10000 <= kind < 20000` OR `0, 3` | The newest event for each `pubkey` and `kind`          |
+| Ephemeral     | `20000 <= kind < 30000`           | Nothing                                                |
+| Parameterized | `30000 <= kind < 40000`           | The newest event for each `pubkey`, `kind` and `d`-tag |
 
-### Kinds
+Parameterized Replaceable events MUST include a `d`-tag with an identifier value to be used as a reference. Tag values that reference the newest replaceable MUST use this format: `<kind>:<32-byte lowercase hex of a pubkey>:<d-tag value>`.
```

Comment:
I like this emphasis on #d (see my confusion here: https://github.com/nostr-protocol/nips/pull/982)